// @vicinae/api/dist/api/clipboard.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.Clipboard = void 0;
const bus_1 = require("./bus");
const clipboard_1 = require("./proto/clipboard");
exports.Clipboard = {
    mapContent(content) {
        let ct = clipboard_1.ClipboardContent.create();
        if (typeof content != "object") {
            ct.text = `${content}`;
        }
        else {
            if (content["file"]) {
                ct.path = { path: content["file"] };
            }
            else if (content["html"]) {
                ct.html = { html: content["html"], text: content["text"] };
            }
            else {
                ct.text = content["text"];
            }
        }
        return ct;
    },
    async copy(text, options = {}) {
        await bus_1.bus.turboRequest("clipboard.copy", {
            content: this.mapContent(text),
            options: { concealed: options.concealed ?? false },
        });
    },
    async paste(text) {
        await bus_1.bus.turboRequest("clipboard.paste", {
            content: this.mapContent(text),
        });
    },
    async read(options) {
        const res = await bus_1.bus.request("clipboard.read", {
            options,
        });
        return res.data.content;
    },
    async readText(options) {
        const res = await bus_1.bus.request("clipboard.read-text", {
            options,
        });
        return res.data.content;
    },
    async clear(text) {
        // TODO: implement
    },
};
